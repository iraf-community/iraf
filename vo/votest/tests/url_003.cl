#
#  Rudimentary URL handling: votables

# Set the test description string.
votest.descr = "Rudimentary URL handling: votables"

# Convert the data$logical to a local path.
s1 = data_url // "/usno-b.xml"
s2 = "file://" // data_path // "/usno-b.xml"
s3 = "file:///localhost" // data_path // "/usno-b.xml"

fcache init

unlearn ("tprint")
tprint.rows="1-5"
tprint.columns="id,sRA,sDec"

# Execute the test commands.
print ("\nLogical Path:  data$usno-b.xml")              # logical path
tprint ("data$usno-b.xml", rows="1-5")

print ("\nHTTP URI:  " // s1)                           # remote http URI
tprint (s1, rows="1-5")

print ("\nFile URI:  " // s2)                           # file URI
tprint (s2, rows="1-5")

print ("\nFile URI:  " // s3)                           # file URI
tprint (s3, rows="1-5")

print ("\nHTTP Redirection")				# HTTP Redirection
tprint ("http://iraf.noao.edu/votest/redir_vot.php", rows="1-5")

print ("\nHTTP CGI query")				# HTTP CGI query
tprint http://iraf.noao.edu/scripts/voget?t=votable&f=usno-b.xml rows="1-5"
tprint ("http://iraf.noao.edu/scripts/voget?t=votable&f=usno-b.xml",rows="1-5")

s1 = "http://iraf.noao.edu/scripts/voget?t=votable&f=usno-b.xml"
tprint (s1, rows="1-5")

