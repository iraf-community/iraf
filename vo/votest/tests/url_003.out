
Logical Path:  data$usno-b.xml
#  Table data$usno-b.xml  Wed 12:17:11 27-Jul-2011

# row id                    sRa        sDec
#                           mas         mas

    1 1369-0259494          999         999
    2 1369-0259503           89          53
    3 1369-0259504          999          16
    4 1369-0259505          486         192
    5 1369-0259508          405         569

HTTP URI:  http://iraf.noao.edu/votest//usno-b.xml
# row id                    sRa        sDec
#                           mas         mas

    1 1369-0259494          999         999
    2 1369-0259503           89          53
    3 1369-0259504          999          16
    4 1369-0259505          486         192
    5 1369-0259508          405         569

File URI:  file:///iraf/iraf/vo/votest/data//usno-b.xml
# row id                    sRa        sDec
#                           mas         mas

    1 1369-0259494          999         999
    2 1369-0259503           89          53
    3 1369-0259504          999          16
    4 1369-0259505          486         192
    5 1369-0259508          405         569

File URI:  file:///localhost/iraf/iraf/vo/votest/data//usno-b.xml
# row id                    sRa        sDec
#                           mas         mas

    1 1369-0259494          999         999
    2 1369-0259503           89          53
    3 1369-0259504          999          16
    4 1369-0259505          486         192
    5 1369-0259508          405         569

HTTP Redirection
# row id                    sRa        sDec
#                           mas         mas

    1 1369-0259494          999         999
    2 1369-0259503           89          53
    3 1369-0259504          999          16
    4 1369-0259505          486         192
    5 1369-0259508          405         569

HTTP CGI query
# row id                    sRa        sDec
#                           mas         mas

    1 1369-0259494          999         999
    2 1369-0259503           89          53
    3 1369-0259504          999          16
    4 1369-0259505          486         192
    5 1369-0259508          405         569
# row id                    sRa        sDec
#                           mas         mas

    1 1369-0259494          999         999
    2 1369-0259503           89          53
    3 1369-0259504          999          16
    4 1369-0259505          486         192
    5 1369-0259508          405         569
# row id                    sRa        sDec
#                           mas         mas

    1 1369-0259494          999         999
    2 1369-0259503           89          53
    3 1369-0259504          999          16
    4 1369-0259505          486         192
    5 1369-0259508          405         569
